### ISSUE TITLE
Extend ClusterWS class

### ISSUE THREAD

**Issue by carmas123** [ClusterWS/ClusterWS#45]
- [ ] Bug
- [x] Question
- [ ] Suggetion
- [ ] Other

In my project (writed in TypeScript) I extend the ClusterWS class like this:

```
class Server extends ClusterWS {
    public app: express.Application;

    /**
     * Constructor.
     *
     * @class Server
     * @constructor
     */
    constructor() {
        super(new INUserOption());

        // create expressjs application
        this.app = App;
    }
}
```

now ho I can do this in 1.5.0 version?

**Comment by carmas123** [comment]
When I try to convert I got always this error:
TypeError: clusterws_1.default is not a constructor

**Comment by goriunov** [comment]
U should check breaking changes i think you do the wrong import of clusterws as in version 1.5.0 u don't have to specify clusterws:

```js
// Old
const ClusterWS = require('clusterws').ClusterWS

// New
const ClusterWS = require('clusterws')
```

the same concept applies for Client version. Just check relaeses as i said in #44  to find out about all main changes which u have to look out.

**Comment by carmas123** [comment]
This is ok but I use typescript to implement my nodeJS server and with the new concept I've TypeError: clusterws_1.default is not a constructor when the type script is compiled and I cannot define new class from ClusterWS, however I solved differently.

**Comment by goriunov** [comment]
Can u show the code how you import ClusterWS. The top part.

**Comment by carmas123** [comment]
const ClusterWS = require("clusterws");
import { Configurations, CustomObject, Listener, ScaleOptions, Socket, TlsOptions, WorkerFunction } from "clusterws";

**Comment by goriunov** [comment]
Just do like this 
```js
import ClusterWS from 'clusterws'
```

You should remove curly braces because now class is exported by default. This should fix the problem.

**Comment by carmas123** [comment]
Sorry but I try this and this is my result when I compile the TypeScript into JS:

```
class Server extends clusterws_1.default {
    /**
     * Constructor.
     *
     * @class Server
     * @constructor
     */
    constructor() {
        super(new INUserOption());
        // create expressjs application
        this.app = App;
    }
}
```

and this is my error:

> class Server extends clusterws_1.default {
>                                  ^
> TypeError: Class extends value undefined is not a constructor or null

**Comment by carmas123** [comment]
I think that this is a bug or bad configuration of my tsconfig.json :(

**Comment by goriunov** [comment]
can u show your `tsconfig.json`

**Comment by carmas123** [comment]
```
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "moduleResolution": "node",
    "sourceMap": true,
    "outDir": "dist",
    "baseUrl": ".",
    "paths": {
      "*": [
        "src/types/*",
        "typings/modules/*",
        "node_modules/*"
      ]
    }
  },
  "include": [
    "src/**/*"
  ]
}
```

**Comment by goriunov** [comment]
Thanks for noticing this one and informing, I have been looking into it and as a result, i have found a fix, update ClusterWS to version 1.5.7. This should fix the error with importing. You should be able to import with 
```js
import ClusterWS from 'clusterws'
```

Also if u use typescript on the front end and have similar problem you can update client version as well.

There are no breaking changes in this release just fix for typescript library.

Thanks again.

**Comment by carmas123** [comment]
You are welcome. Thanks
